linters/errorsnew/errorsnew_linter.go

```diff
@@ -0,0 +1,40 @@
+package errorsnew
+
+import (
+	"go/ast"
+	"go/token"
+	"golang.org/x/tools/go/analysis"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "errorsnew",
+	Doc:  "checks that there are no calls to errors.New outside of a function (we should use sentinel errors for that)",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
```

Comment:
I love the linter, but don't love the level of nesting. 
Add some functions to reduce nesting level (e.g. `func inspectFile(pass, file)`, `func inspectValueSpec(...)` etc)